packages/wdio-jasmine-framework/src/reporter.js

```diff
@@ -87,10 +98,47 @@ export default class JasmineReporter {
             })
         }
 
+        // errors on beforeAll are swallowed by Jasmine and can only be accessed later on
+        let errorsBeforeAll
+        let errorsAfterAll
+        if (suite.failedExpectations && suite.failedExpectations.length) {
```

Comment:
suite.failedExpectations is always an array

Reply:
Is it guaranteed that Jasmine will always provide it? I was not sure, which is why I added the check, but I could also remove it

Reply:
According to their docs on custom reporter it should be

Reply:
Just removed it and added empty `failedExpectation` arrays to each of the unit tests as well with `suiteDone` so they won't fail